pandas/_libs/lib.pyx

```diff
@@ -750,7 +750,11 @@ cpdef ndarray[object] ensure_string_array(
 
     if hasattr(arr, "to_numpy"):
 
-        if hasattr(arr, "dtype") and arr.dtype.kind in "mM":
+        if (
+            hasattr(arr, "dtype")
+            and arr.dtype.kind in "mM"
+            and not hasattr(arr, "_pa_array")
```

Comment:
This is a bit ugly, but essentially this is avoiding that `astype(str)` for ArrowExtensionArray (if that happens to have a datetimelike dtype) is called here, because that will use `ensure_string_array` for that implementation, causing a recursion error.

An alternative would be a better check than `arr.dtype.kind in "mM"` that restricts itself to `DatetimeLikeArrayMixin` (essentially I need a `isinstance(arr, DatetimeLikeArrayMixin)`, I think). Could potentially do that with some ABC check.

Another alternative is to handle `astype(str)` specifically in ArrowExtensionArray (currently `astype` is not implemented there, and it falls back entirely on the base class implementation)

Reply:
Added a TODO comment